lib/widget/confirmation_input.dart

```diff
@@ -0,0 +1,152 @@
+      'activeFieldBorderColor': (newValue) =>
+          _controller.activeFieldBorderColor = Utils.getColor(newValue),
+      'defaultFieldBackgroundColor': (newValue) =>
+          _controller.defaultFieldBackgroundColor = Utils.getColor(newValue),
+      'activeFieldBackgroundColor': (newValue) =>
+          _controller.activeFieldBackgroundColor = Utils.getColor(newValue),
+      'filledFieldBackgroundColor': (newValue) =>
+          _controller.filledFieldBackgroundColor = Utils.getColor(newValue),
+      'filledFieldBorderColor': (newValue) =>
+          _controller.filledFieldBorderColor = Utils.getColor(newValue),
+      'cursorColor': (newValue) =>
+          _controller.cursorColor = Utils.getColor(newValue),
+      'onChange': (funcDefinition) => _controller.onChange =
+          EnsembleAction.fromYaml(funcDefinition, initiator: this),
+      'onSubmit': (funcDefinition) => _controller.onSubmit =
+          EnsembleAction.fromYaml(funcDefinition, initiator: this),
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
+
+  @override
+  ConfirmationInputState createState() => ConfirmationInputState();
+}
+
+class ConfirmationInputController extends BoxController {
+  String? text;
+  int? length;
+  bool? enableCursor;
+  String? fieldType;
+  Color? defaultFieldBorderColor;
+  Color? activeFieldBorderColor;
+  Color? defaultFieldBackgroundColor;
+  Color? activeFieldBackgroundColor;
+  Color? filledFieldBackgroundColor;
+  Color? filledFieldBorderColor;
+  Color? cursorColor;
+  EnsembleAction? onChange;
+  EnsembleAction? onSubmit;
+
+  TextStyleComposite? _textStyle;
+  TextStyleComposite get textStyle => _textStyle ??= TextStyleComposite(this);
+  set textStyle(TextStyleComposite style) => _textStyle = style;
+}
+
+class ConfirmationInputState extends framework.WidgetState<ConfirmationInput> {
+  @override
+  Widget buildWidget(BuildContext context) {
+    return BoxWrapper(
+      widget: buildTextInput(widget.controller),
+      boxController: widget.controller,
+    );
+  }
+
+  Widget buildTextInput(ConfirmationInputController controller) {
+    return OtpPinField(
+      otpPinFieldStyle: OtpPinFieldStyle(
+        textStyle: controller.textStyle.getTextStyle(),
+        defaultFieldBorderColor:
+            controller.defaultFieldBorderColor ?? Colors.black45,
+        activeFieldBorderColor:
+            controller.activeFieldBorderColor ?? Colors.black,
+        defaultFieldBackgroundColor:
+            controller.defaultFieldBackgroundColor ?? Colors.transparent,
+        activeFieldBackgroundColor:
+            controller.activeFieldBackgroundColor ?? Colors.transparent,
+        filledFieldBackgroundColor:
+            controller.filledFieldBackgroundColor ?? Colors.transparent,
+        filledFieldBorderColor:
+            controller.filledFieldBorderColor ?? Colors.transparent,
+      ),
+      maxLength: controller.length ?? 4,
```

Comment:
don't assume 4 here. If it's a required field, expose it.

Reply:
It's not a required field. So I length as a late property and in the setters, I set it as a 4 for the fallback value.